lua/telescope/pickers/entry_display.lua

```diff
@@ -8,11 +10,19 @@ entry_display.create = function(configuration)
   for _, v in ipairs(configuration.items) do
     if v.width then
       local justify = v.right_justify
+      local width
       table.insert(generator, function(item)
+        if width == nil then
+          local status = state.get_status(vim.api.nvim_get_current_buf())
+          local s = {}
+          s[1] = vim.api.nvim_win_get_width(status.results_win) - #status.picker.selection_caret
+          s[2] = vim.api.nvim_win_get_height(status.results_win)
+          width = resolve.resolve_width(v.width)(nil, s[1], s[2])
+        end
```

Comment:
this should not work. we need a table because multiple functions point to the same reference. So the width for the second column will always be the same as the width for the first column. (Under the assumption that the second one has a width set)

Reply:
I didn't think that this was the case, since `width` is scoped inside the for loop.
In my tests things seem to work as expected, although I might not be testing widely enough.

Reply:
You are right